src/plugins/discover/common/utils/sorting/get_sort_for_search_source.ts

```diff
@@ -31,16 +40,31 @@ export function getSortForSearchSource(
       return [{ _score: defaultDirection } as EsQuerySortValue];
     }
   }
+
   const { timeFieldName } = dataView;
-  return getSort(sort, dataView).map((sortPair: Record<string, string>) => {
-    if (dataView.isTimeNanosBased() && timeFieldName && sortPair[timeFieldName]) {
-      return {
-        [timeFieldName]: {
-          order: sortPair[timeFieldName],
-          numeric_type: 'date_nanos',
-        },
-      } as EsQuerySortValue;
+  const sortPairs = getSort(sort, dataView);
+
+  const sortForSearchSource = sortPairs.map((sortPair: Record<string, string>) => {
+    if (timeFieldName && sortPair[timeFieldName]) {
+      return getESQuerySortForTimeField(
+        timeFieldName,
+        sortPair[timeFieldName] as SortDirection,
+        dataView.isTimeNanosBased()
+      );
     }
     return sortPair as EsQuerySortValue;
   });
+
+  // TODO: do we need to have a tie breaker like this?
```

Comment:
Would it be a tie breaker like mentioned in this Issue ? https://github.com/elastic/kibana/issues/47934

Reply:
@kertal Yes, in this implementation I am using a tie breaker as it's done in context app: based on Advanced Setting `context:tieBreakerFields`. I think it makes sense to add it but this change can be considered as a breaking change (also for CSV). So I would need your advice on whether to add it now or not.

Reply:
Good point about the breaking change. Is it required or optional for the implementation? In terms of breaking change, it don't think it would break existing functionality inside Discover, but, yes, mainly it's about CSV, it could lead to a different sort order when exporting and the tie breaker is not implemented.

Would it be an option to exclude the tiebreaker for this PR and implement it in a follow up / later, with also CSV included?

Reply:
@kertal Current implementation of CSV relies on Discover search source `sort` param and reuses it's value. That means that if a tie breaker is not used for Discover then it's not used for CSV generation at the moment. @tsullivan please correct if my assumption is wrong. It's a good question if an explicit tie breaker is required or not.

Reply:
just to make sure, since we continued the discussion "offline" 
@jughosta mentioned
> CSV export is done with search_after and pit so I guess according to docs it is running with _shard_doc by default

So if we introduce a tiebreaker for Discover like `_doc`  that we use for incremental data fetching, this would not influence CSV generation, on the contrary, it would make the data fetching more similar, since we'd also use a tie breaker. Can you confirm @tsullivan @rshen91 ? Many thx!

Reply:
Quick update here: added `_doc` as a tie breaker for Discover only so far. It would not affect CSV at this point.